src/main/java/com/bunq/sdk/http/ApiClient.java

```diff
@@ -156,24 +151,16 @@ private void setProxyIfNeeded(OkHttpClient.Builder httpClientBuilder) {
     }
   }
 
-  /**
-   */
-  private CertificatePinner determineCertificateToPin() {
-    CertificatePinner.Builder certificateBuilder = new CertificatePinner.Builder();
-
-    switch (apiContext.getEnvironmentType()) {
-      case PRODUCTION:
-        return certificateBuilder.add(
-            apiContext.getBaseUri(), PINNED_KEY_PRODUCTION
-        ).build();
-      case SANDBOX:
-        return certificateBuilder.add(
-            apiContext.getBaseUri(), PINNED_KEY_SANDBOX
-        ).build();
-      default:
-        throw new BunqException(
-            String.format(ERROR_AMI_ENVIRONMENT_NOT_EXPECTED, apiContext.getEnvironmentType().toString())
-        );
+  private static CertificatePinner determineCertificateToPin(ApiEnvironmentType environmentType) {
+    if(environmentType != null && environmentType.getPinnedKey() != null) {
+      return new CertificatePinner.Builder().add(
+              environmentType.getBaseUri(), environmentType.getPinnedKey()
+      ).build();
+    } else {
+      throw new BunqException(
+              String.format(ERROR_AMI_ENVIRONMENT_NOT_EXPECTED, 
+				  Objects.toString(environmentType, "<null>"))
+      );
     }
   }
```

Comment:
Could you be kind enough to add the missing new line at end of the `class` block here as well 👼